### ISSUE TITLE
How to check if vehicle is sleeping or awake

### ISSUE THREAD

**Issue by Moosbueffel** [onokje/node-red-contrib-tesla#38]
How can I check if the vehicle is sleeping or awake? Perhaps it the temp == null?? But this seems me a workaround. 

Could you please implement a command for this? I think it is always implemented in your code but not wired ouside!?

Thanks

LInks:
https://support.teslafi.com/en/knowledge-bases/2/articles/640-enabling-sleep-settings-to-limit-vampire-loss
https://www.domoticz.com/forum/viewtopic.php?p=240152#p240152

**Comment by takipo** [comment]
You can send the "vehicle" (retrieve general vehicle information) command and check for msg.payload.state == online. It will send == asleep when its sleeping.

**Comment by Moosbueffel** [comment]
Many thanks, I will try it, when I'm back home next week and give feedback.

**Comment by Moosbueffel** [comment]
Ok, danke.
Dann sind die Werte die bei _vehicle_ geliefert werden die vom letzten Wachzustand? Und Fahrzeug wird  durch diesen Aufruf nicht aufgeweckt? Denn aufwecken will ich das Auto ja nicht.

**Comment by Michael-K-at-GitHub** [comment]
Ja, aber bei (vehicles) wird selbst ja nicht sehr viel zurückgeliefert. Nur welche Fahrzeuge du hast und ob diese wach sind.

Wenn Du den Batteriestand et al dauerhaft brauchst, musst du ihn dir merken. Weil danach fragen (vehicleData oder chargeState) weckt das Auto auf (weil er ja fragen muss, wie der Akku jetzt steht).

Was ich mache ist relativ simpel: 
`tesla-command:vehicles --> switch node (msg.payload[0].state == online -> Output 1 / Otherwise -> Output 2)`

an Output 1
`get VehicleData --> change node: set flow.TeslaDataBackup to to value msg.payload; msg.Online to the value true`

an Output 2
`change node: set msg.payload to the value flow.TeslaDataBackup; msg.Online to the value false`

dann die beiden flows zusammenführen und du hast in msg die aktuellen Daten und in msg.Online zusätzlich den Status ob wach oder nicht

**Comment by Michael-K-at-GitHub** [comment]
Ich lasse obige Sequenz alle 60 Sekunden laufen. Trotzdem schläft er zuverlässig nach 12 Minuten ein.

**Comment by Moosbueffel** [comment]
> Ich lasse obige Sequenz alle 60 Sekunden laufen. Trotzdem schläft er zuverlässig nach 12 Minuten ein.

Ok, muß ich mal ausprobieren. Man soll ja auch nicht zu oft abfragen. Alle Minute, wäre natürlich gut

**Comment by Michael-K-at-GitHub** [comment]
Es gibt jedenfalls eine Limitierung, aber 1/Minute funktioniert bei mir weit Monaten zuverlässig.
